### ISSUE TITLE
MapAccess/SeqAccess: how to pass owned values instead of T: DeserializeSeed<'de>? 

### ISSUE THREAD

**Issue by pantonov** [serde-rs/serde#1774]
When decoding sequences in my format, I need first to decode byte sequence in a special way, and then to use another temporary `Deserializer` instance while deserializing a sequence. However, trait methods such as `next_element_seed` in `SeqAccess/MapAccess` traits bind the lifetime of values
to the lifetime of `Deserializer` by using trait bound `T: DeserializeSeed<'de>`, so apparently these methods cannot use values produced by another `Deserializer` instance. Is there any workaround for this?

In other words, when we take typical deserialization code like
```rust
struct SeqAccess<'a, 'de: 'a, R> {
    deserializer: &'a mut Deserializer<'de, R>,
}
```
and try to set deserializer to anything except `&mut *self` and pass this instance to visitor.visit_seq, compiler will complain.

Any suggestions?

**Comment by pantonov** [comment]
Edit: solved by removing `'de` lifetime from Deserializer struct by parameterizing it by generic Reader which can hold either reference to input data or buffer.
